<?xml version="1.0"?>
<launch>
    <param name="maxlen" type="double" value="100.0" />
    <param name="minlen" type="double" value="0.0" />
    <param name="max_ang_FOV" type="double" value="2.7" />
    <param name="min_ang_FOV" type="double" value="0.5" />
    <param name="pcTopic" type="str" value="/velodyne_points" />
    <param name="imgTopic" type="str" value="/usb_cam/image_raw" />
    <param name="x_resolution" type="double" value="0.25" />
    <param name="y_interpolation" type="double" value="10.0" />
    <param name="ang_Y_resolution" type="double" value="2.05" />
    <param name="max_var" type="double" value="5.0" />
    <param name="filter_output_pc" type="bool" value="true" />
    <param name="num_layers" type="int" value="16" />
    <param name="min_window_size" type="int" value="3" />
    <param name="max_window_size" type="int" value="9" />

    <group ns="matrix_file">
        <rosparam file="$(find lidar_camera_fusion)/cfg/cfg_params.yaml" />
    </group>

    <node pkg="lidar_camera_fusion" type="lidar_camera_node" name="lidar_camera_node" output="screen" />

    <!-- Velodyne -->
    <include file="$(find velodyne_pointcloud)/launch/VLP16_points.launch" />

    <!-- RViz (optional, uncomment if needed) -->
    <!-- <node type="rviz" name="rviz" pkg="rviz" args="-d $(find lidar_camera_fusion)/rviz/lidar_camera.rviz" /> -->
</launch>
